recipes/newrelic/infrastructure/ohi/mongodb/debian.yml

```diff
@@ -62,24 +59,344 @@ preInstall:
       > db.createUser({ user: "username", pwd: "password", roles: [ "clusterMonitor", "listCollections" ]})
 
   requireAtDiscovery: |
-    MONGODB_EXISTS=$(mongo --eval 'db.runCommand({ connectionStatus: 1 })' | grep "ok" | wc -l)
-    if [ $MONGODB_EXISTS -eq 0 ]; then
-      exit 132
-    fi
-    exit 1
+    # TODO: Fix this. If Auth or SSL enable, this required discovery step fails
+    # MONGODB_EXISTS=$(mongo --eval 'db.runCommand({ connectionStatus: 1 })' | grep -e '\"ok\" : 1' | wc -l)
+    # if [ $MONGODB_EXISTS -eq 0 ]; then
+    #   exit 132
+    # fi
+    exit 0
 
 install:
   version: "3"
   silent: true
   tasks:
     default:
       cmds:
-        - task: setup
+        - task: assert_pre_req
+        - task: mongo_pre_req
+        - task: assert_input
+          vars:
+            MAX_RETRIES: 3
+        - task: restart
+        - task: cleanup
+
+    assert_pre_req:
+      cmds:
+        - |
+          IS_INFRA_INSTALLED=$(sudo ps aux | grep newrelic-infra-service | grep -v grep | wc -l)
+          if [ $IS_INFRA_INSTALLED -eq 0 ] ; then
+            echo "The infrastructure agent is required to install this integration, we recommend going through our guided install path for this pre-requisite which can be found here:  https://docs.newrelic.com/docs/full-stack-observability/observe-everything/get-started/new-relic-guided-install-overview" >&2
+            exit 1
+          fi
+        - |
+          # Map of tool names to the associated error code
+          REQUIRED_TOOLS_AND_ERROR_CODES="date:41 sha256sum:42 base64:43 head:44"
+
+          for tuple in $REQUIRED_TOOLS_AND_ERROR_CODES; do
+            tool=$(echo ${tuple} |cut -d':' -f1)
+            code=$(echo ${tuple} |cut -d':' -f2)
+
+            IS_TOOL_INSTALLED=$(which ${tool} | wc -l)
+            if [ "$IS_TOOL_INSTALLED" -eq 0 ]
+            then
+              echo -e "{{.RED}}This installation requires '${tool}' to be installed.{{.NOCOLOR}}" >> /dev/stderr
+              exit ${code}
+            fi
+          done
+        - | 
+          IS_MONGO_SHELL_INSTALLED=$(which mongo | wc -l)
+          if [ $IS_MONGO_SHELL_INSTALLED -eq 0 ]; then
```

Comment:
@amolero-nr this might need to go in the pre-req. If we cant create a username/password we wont be able to install. We should still detect though.

Reply:
Ok